### PR TITLE
Enemy Scaling and Rebalance

### PR DESCRIPTION
**Pull request by AntonAsbrink**
Made so zombie get stronger with each level and stats has been adjusted on player and upgrades to make it more difficult

### CODE REVIEW

**Review by kattalainen — COMMENTED** [review]
A lot of small changes, just make sure you test to see that things work (upgrades/leveling, prices etc) :> 
I tested it for half an hour straight and it gets very difficult quite fast! (even when I upped my damage to 100 and health to 500 for testing). 
One small nitpick is that in the shop I would like to see what level of the upgrade I am buying, rather than what level of the upgrade I already have.

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 1]
```diff
@@ -201,6 +200,7 @@ Entity* EnemyFactory::Add(const std::string& entityName, EnemyComps* comps)
 	t->SetRotationY(comps->rot.y);
 	t->SetRotationZ(comps->rot.z);
 	t->SetVelocity(comps->movementSpeed * 0.5);
+	t->UpdateWorldMatrix();
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:203]: Does this fix the "spawn huge at origo" bug?

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 2]
```diff
@@ -414,15 +414,17 @@ void EnemyFactory::onSceneSwitch(SceneChange* evnt)
...
-		m_LevelMaxEnemies = 20 + log(m_DifficultScale) * 10;
-
 		Entity* enemyGui = m_pScene->GetEntity("enemyGui");
 		if (enemyGui != nullptr)
 		{
 			enemyGui->GetComponent<component::GUI2DComponent>()->GetTextManager()->SetText("Enemies: 0/" + std::to_string(m_LevelMaxEnemies), "enemyGui");
 		}
 
+		//Scaling difficulty
+		m_LevelMaxEnemies += 2;
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: Move this up above the if-statement

**Review by kattalainen — DISMISSED** [review]

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
Fix the stuff

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp (hunk 1 above)
Comment (AntonAsbrink) [reply StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp:203]: me think so maybe

**Review by LeoWikstrom — APPROVED** [review]
#fixedthestuff 👍

**Review by kattalainen — APPROVED** [review]

**Pull request merged**